app/src/main/kotlin/com/simplemobiletools/applauncher/adapters/LaunchersAdapter.kt

```diff
@@ -27,8 +27,12 @@ import kotlinx.android.synthetic.main.item_launcher_label.view.*
 import java.util.*
 
 class LaunchersAdapter(
-    activity: SimpleActivity, val launchers: ArrayList<AppLauncher>, val listener: RefreshRecyclerViewListener?,
-    recyclerView: MyRecyclerView, itemClick: (Any) -> Unit
+    activity: SimpleActivity,
+    val launchers: ArrayList<AppLauncher>,
+    val listener: RefreshRecyclerViewListener?,
+    recyclerView: MyRecyclerView,
+    var onItemsRemoved: (() -> Unit)? = null,
```

Comment:
this is a way overcomplicated solution too, just reuse RefreshRecyclerViewListener from Simple Commons, like at Simple Calendars EventListAdapter or some other one. Call it only if all items are deleted.

Reply:
> reuse RefreshRecyclerViewListener

you mean to call `refreshItems()` when the list is emptied, right?

or do you wanna create another listener with `onItemsCleared()`? (a little more efficient but doesn't matter in this case)